tachyon/math/matrix/matrix_utils.h

```diff
@@ -155,7 +160,14 @@ void ReverseMatrixIndexBits(Eigen::MatrixBase<Derived>& mat) {
   OMP_PARALLEL_FOR(size_t row = 1; row < rows; ++row) {
     size_t ridx = base::bits::ReverseBitsLen(row, log_n);
     if (row < ridx) {
-      mat.row(row).swap(mat.row(ridx));
+      auto row1 = absl::MakeSpan(
+          reinterpret_cast<uint8_t*>(mat.derived().data() + row * mat.cols()),
+          mat.cols() * sizeof(typename Derived::Scalar));
+      auto row2 = absl::MakeSpan(
+          reinterpret_cast<uint8_t*>(mat.derived().data() + ridx * mat.cols()),
+          mat.cols() * sizeof(typename Derived::Scalar));
```

Comment:
```suggestion
      absl::Span<uint8_t> row1(
          reinterpret_cast<uint8_t*>(mat.derived().data() + row * mat.cols()),
          mat.cols() * sizeof(typename Derived::Scalar));
      absl::Span<uint8_t> row2(
          reinterpret_cast<uint8_t*>(mat.derived().data() + ridx * mat.cols()),
          mat.cols() * sizeof(typename Derived::Scalar));
```

Reply:
Please add something like `if constexpr (Derived::Options == Eigen::RowMajor)`. I am not sure about the condition, please take a look at the code in Eigen matrix.